scripts/benchmark_all.sh

```diff
@@ -0,0 +1,152 @@
+#!/bin/bash
+
+set -e
+
+if [ $# -ne 2 ]
+then
+  echo 'This script is used to compare the performance impact of a change. Running it takes more than an hour.'
+  echo '  It compares two git revisions using various benchmarks (see below) and prints the result in a format'
+  echo '  that is copyable to Github.'
+  echo 'Typical call (in a release build folder): ../scripts/benchmark_all.sh master HEAD'
+  exit 1
+fi
+
+benchmarks='hyriseBenchmarkTPCH hyriseBenchmarkTPCDS hyriseBenchmarkTPCC hyriseBenchmarkJoinOrder'
+num_mt_clients=50
+
+# Retrieve SHA-1 hashes from arguments (e.g., translate "master" into an actual hash)
+start_commit_reference=$1
+end_commit_reference=$2
+shift; shift
+start_commit=$(git rev-parse $start_commit_reference | head -n 1)
+end_commit=$(git rev-parse $end_commit_reference | head -n 1)
+
+# Check status of repository
+if [[ $(git status --untracked-files=no --porcelain) ]]
+then
+  echo 'Cowardly refusing to execute on a dirty workspace'
+  exit 1
+fi
+
+# Ensure that this is run on a release build. Not really necessary, just a sanity check.
+output=$(grep 'CMAKE_BUILD_TYPE:STRING=Release' CMakeCache.txt || true)
+if [ -z "$output" ]
+then
+  echo 'Current folder is not configured as a release build'
+  exit 1
+fi
+
+# Check whether to use ninja or make
+output=$(grep 'CMAKE_MAKE_PROGRAM' CMakeCache.txt | grep ninja || true)
+if [ ! -z "$output" ]
+then
+  build='ninja'
+else
+  build='make'
+fi
+
+# Create the result folder if necessary
+mkdir benchmark_all_results 2>/dev/null || true
+
+build_folder=$(pwd)
+
+# Here comes the actual work
+for commit in $start_commit $end_commit
+do
+  # Skip commits where we already have all results
+  if [ -f "benchmark_all_results/complete_${commit}" ]
+  then
+    echo "Benchmarks have already been executed for ${commit}. Skipping."
+    echo "Run rm benchmark_all_results/*${commit}* to re-execute."
+    sleep 1  # Easiest way to make sure the above is read
+    continue
+  fi
+
+  # Checkout and build from scratch, tracking the compile time
+  git checkout $commit
+  git submodule update --init
+  echo "Building $commit..."
+  $build clean
+  /usr/bin/time -p sh -c "( $build -j $(nproc) ${benchmarks} 2>&1 ) | tee benchmark_all_results/build_${commit}.log" 2>"benchmark_all_results/build_time_${commit}.txt"
```

Comment:
`nproc` is a Linux-only utility. The equivalent on macOS is `sysctl -n hw.logicalcpu`

Reply:
How about adding coreutils to the brew requirements?

Reply:
Might be overkill for this one instance, but since that is one of the most useful packets and provides compatibility for a lot of GNU tools, it might be a good idea anyway.

Reply:
Ok for me. But it should be part of the install-script.